VTE evaluate measure

tstrass opened the pull request:
We don't yet have a cqf-ruler instance that we know for sure is evaluating the measure correctly, but if you use the instance on @okeefm's machine, it should at least return an OK response, but all populations in the measure report will be 0. So for now this doesn't make assertions on the populations, but it should in the future.

Run with:
```
brake crucible:execute[<base stu3 url>,stu3,vte-evaluate-measure]
```

tstrass committed:
Add TestScript for VTE evaluate-measure

tstrass committed:
Add fixtues for VTE evaulate measure TestScript

tstrass committed:
Add teardown for all resources in VTE evaluate measure TestScript

okeefm commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+                <description value="Evaluate measure with server assigned resource id."/>
+                <accept value="json"/>
+                <contentType value="json"/>
+                <url value="Measure/${createMeasureId}/$evaluate-measure?periodStart=${periodStart}&amp;periodEnd=${periodEnd}"/>
```
Do we want to change this `url` field to be the `params` field, like in https://github.com/fhir-crucible/plan_executor/blob/fhir_prototype_dev/lib/tests/testscripts/scripts/reporting/vte-submit-data.xml#L240 ? That just makes us more compliant with how Aegis writes their `TestScript`s, in case we want to upload it to their server later on.

tstrass replied:
good call

mgramigna commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+        </action>
+        <action>
+            <assert>
+                <description value="Confirm that the returned HTTP status is 201(Created)."/>
```
Since this assertion is looking for either 200 or 201, the description should reflect that

mgramigna commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+        </action>
+        <action>
+            <assert>
+                <description value="Confirm that the returned HTTP status is 201(Created)."/>
```
Since this assertion is looking for either 200 or 201, the description should reflect that

mgramigna commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+        </action>
+        <action>
+            <assert>
+                <description value="Confirm that the returned HTTP status is 201(Created)."/>
```
Since this assertion is looking for either 200 or 201, the description should reflect that

mgramigna commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+                    <code value="delete"/>
+                </type>
+                <resource value="Encounter"/>
+                <description value="Delete the Encounter from $submit-data"/>
```
Update description to not mention $submit-data

mgramigna commented on lib/tests/testscripts/scripts/reporting/vte-evaluate-measure.xml:
```diff
@@ -0,0 +1,413 @@
+                    <code value="delete"/>
+                </type>
+                <resource value="Patient"/>
+                <description value="Delete the Patient from $submit-data"/>
```
Update description to not mention $submit-data

mgramigna reviewed (changes requested):
This looks good. I have some small suggestions

* The comments I made to update the descriptions of some of the assertions
* We might want to add an assertion to check that the response is a MeasureReport, as this is required by definition of the $evaluate-measure operation

tstrass committed:
Add MeasureReport assertion in VTE evaluate-measure TestScript

Also make a few documentation adjustments in the TestScript per review
comments.

tstrass committed:
Modify evaluate-measure call in TestScript to use params instead of url

okeefm reviewed (approved):
Looks good to me.
